src/main/groovy/com/github/prokod/gradle/crossbuild/ResolutionStrategyConfigurer.groovy

```diff
@@ -69,46 +72,49 @@ class ResolutionStrategyConfigurer {
             def dependencySets = insightsView.getDependencySets(DependencySetType.ALL)
 
             def allDependencies = dependencySets.flatMapped()
-            project.logger.info(LoggerUtils.logTemplate(project,
-                    lifecycle:'afterEvaluate',
-                    configuration:crossBuildConfigurationName,
-                    parentConfiguration:names.main,
-                    msg:"Inherited dependendencies to consider while resolving ${crossBuildConfigurationName} " +
-                            'configuration dependencies: [' +
-                            "${allDependencies.collect { "${it.group}:${it.name}" }.join(', ')}]"
-            ))
 
+            //avoid big string creation
+            if (project.logger.infoEnabled) {
+                project.logger.info(LoggerUtils.logTemplate(project,
+                    lifecycle: 'afterEvaluate',
+                    configuration: crossBuildConfigurationName,
+                    parentConfiguration: names.main,
+                    msg: "Inherited dependendencies to consider while resolving ${crossBuildConfigurationName} " +
+                        'configuration dependencies: [' +
+                        "${allDependencies.collect { "${it.group}:${it.name}" }.join(', ')}]"
+                ))
+            }
             def di = new DependencyInsights(sourceSetInsights)
 
-            def crossBuildProjects = CrossBuildPluginUtils.findAllCrossBuildPluginAppliedProjects(insightsView)
```

Comment:
Why did you remove that ?

Reply:
this was the main problem that for each project you run this funciton and then search all other projects with cross build enabled and then configure the rules for each project with crossbuild enabled.

i.e. 30 projects find the other 30 projects which results in 900 times that these are configured while it only needs to be configured 30 times.